app/src/ui/history/commit-summary.tsx

```diff
@@ -293,9 +315,20 @@ export class CommitSummary extends React.Component<
     )
   }
 
-  public render() {
-    const shortSHA = this.props.commit.shortSha
+  private getShaRef = (useShortSha?: boolean) => {
+    const { commits } = this.props
+    const oldest = useShortSha ? commits[0].shortSha : commits[0].sha
+
+    if (commits.length === 1) {
+      return oldest
+    }
+
+    const latest = useShortSha ? commits.at(-1)?.shortSha : commits.at(-1)?.sha
```

Comment:
Nit: small refactor here?
```suggestion
    const latestCommit = commits.at(-1)
    const latest = useShortSha ? latestCommit?.shortSha : latestCommit?.sha
```